lib/ansible/modules/packaging/os/bindep.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/python
+
+# Copyright 2017 Red Hat, Inc.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: bindep
+short_description: Get lists of missing distro packages to be installed
+author: "Monty Taylor (@mordred)"
+version_added: "2.3"
+description:
+    - Retreive a list of packages that are not already installed based on a
+      list of requirements in bindep format. For more information on bindep,
+      please see https://docs.openstack.org/infra/bindep/
+notes:
+    - This module creates a new top-level C(bindep_packages) fact, which
+      contains the list of packages that are not installed.
+requirements:
+    - "python >= 2.7"
+    - "bindep >= 2.2.0"
+options:
+   name:
+     path:
+       - Path to a bindep.txt file or to a directory that contains either
+         a bindep.txt file or an other-requirements.txt file that should
+         be used as input. Mutually exclusive with requirements.
+     required: false
+     default: None
+   requirements:
+     description:
+       - A list of bindep requirements strings. Mutually exclusive with path.
+     required: false
+     default: None
+   profiles:
+     description:
+       - An explicit list of profiles to filter by
+     required: false
+     default: ['default']
+'''
+
+EXAMPLES = '''
+# Get the list of packages to install on a given host from a file
+- bindep:
+    path: /home/example/bindep.txt
+
+# Get the list of packages to install and install them
+- bindep:
+    path: /home/example/bindep.txt
+- package:
+    name: '{{ item }}'
+    state: latest
+  with_items: bindep_packages
```

Comment:
Do we like this as a fact? Or would it be better for this to just return and require a register? (The module has no use other than returning data, so it seemed more like a fact type thing to me, but the example maybe looks a little odd?)